etsin_finder/frontend/js/components/qvain/views/datasets/table.jsx

```diff
@@ -212,6 +212,17 @@ class DatasetTable extends Component {
     this.props.history.push(getQvainUrl(`/dataset/${dataset.identifier}`))
   }
 
+  handleUseAsTemplate = dataset => {
+    const { getQvainUrl } = this.props.Stores.Env
+    this.props.history.push(getQvainUrl('/dataset'))
+
+    if (dataset.next_draft?.identifier) {
```

Comment:
Metax v2 api returns something like `{ next_draft: { identifier: 'some_dataset_identifier' }, ... }`, and Etsin then inserts the actual associated draft dataset there.

Metax v1 api shouldn't return next_draft, but I think at least at some point there was a bug where it might return `{ next_draft: 'some_dataset_identifier', ... }`. This tries to make sure next_draft is actually the dataset object, not just a string. I agree that this probably isn't the cleanest way to do this.